upgrade/upgrade-scripts/upgrade-container

```diff
@@ -15,6 +15,13 @@
 # limitations under the License.
 #
 
+#
+# set umask to 0022 so that all new directories will be created with correct
+# permissions. This is especially required for not-in-place upgrades. This will
+# override the umask set by mgmt stack.
+#
+umask 0022
```

Comment:
I believe this is already the default umask, so is part of the problem the fact that the app-gate changes this?

I think I might agree with Pavel, that we could/should apply this more generally by placing it in `common.sh`.. or alternatively, we could have a more targeted fix, e.g. by making a change in `mount_upgrade_container_dataset` where I believe the problematic `mkdir` is called.

Reply:
Yes the issue is due to the umask being set when calling these scripts. I will move it to common.sh.

Reply:
Please be careful with such an approach. Countless bugs have been introduced by code that has set umask globally under the assumption that it was the "correct" umask for everyone, only to find out that some script implicitly depended on some umask that it had inherited by accident. This is not without risk.

Reply:
Yea. I think, in general, I'd rather we make the logic independent of w/e the umask is.. so if we need something specific, it may be better to target that specific `mkdir` call to work regardless of what the umask is.

But, I'm good with whatever works.. if going back to the default umask, works, I'm OK with that.. but comes with it's own set of risk(s).

Reply:
@sebroy yes that is a risk, but in this particular case, I intend to set umask to 0022 which is the bash default. I would hope that any new script/statements that require a different umask would explicitly set it themselves. Is that a reasonable assumption? This particular common.sh is used by the upgrade scripts only.

Reply:
I think for most platform-related scripts it is assumed that umask would be `0022`, whereas the app-gate is a special more restrictive case.